packages/grpc-js/src/server-call.ts

```diff
@@ -559,52 +544,89 @@ export class Http2ServerCallStream<
     return metadata;
   }
 
-  receiveUnaryMessage(encoding: string): Promise<RequestType> {
-    return new Promise((resolve, reject) => {
-      const stream = this.stream;
-      const chunks: Buffer[] = [];
-      let totalLength = 0;
+  receiveUnaryMessage(
+    encoding: string,
+    next: (
+      err: Partial<ServerStatusResponse> | null,
+      request?: RequestType
+    ) => void
+  ): void {
+    const { stream } = this;
+
+    let receivedLength = 0;
+    const call = this;
+    const body: Buffer[] = [];
+    const limit = this.maxReceiveMessageSize;
+
+    stream.on('data', onData);
+    stream.on('end', onEnd);
+    stream.on('error', onEnd);
+
+    function onData(chunk: Buffer) {
+      receivedLength += chunk.length;
```

Comment:
This changes `byteLength` to `length`. I'm not sure exactly what the difference is, but I think this may be causing the out of bounds access error in one of the tests.